stairway/src/main/java/bio/terra/stairway/FlightMap.java

```diff
@@ -55,130 +46,114 @@ public FlightMap() {
+
+    // If we have an empty list AND there is valid JSON, this predates the flightworking table...
+    // use the json in this case.
+    if (inputList.isEmpty() && json != null) {
+      FlightMap map = new FlightMap();
+      map.fromJson(json);
+      return map;
     }
 
-    return Optional.of(map);
-  }
-
-  /**
-   * Validate whether the passed inputList will deserialize to the same set of keys and
-   * corresponding types stored in the map. Note that this does not ensure the equality of values as
-   * all types stored in FlightMap are not required to be comparable.
-   *
-   * <p>Throws {@link RuntimeException} if any key exists in inputList but not in map (or vice
-   * versa). Throws {@link JsonProcessingException} if any value in inputList does not deserialize
-   * to the type of its corresponding entry in the map.
-   *
-   * @param inputList
-   */
-  @VisibleForTesting
-  void validateAgainst(List<FlightInput> inputList) throws Exception {
-
-    if (inputList.size() != map.size())
-      throw new RuntimeException(
-          String.format(
-              "Passed input list has %d entries, map has %d.", inputList.size(), map.size()));
-
-    for (FlightInput input : inputList) {
-      final String key = input.getKey();
-      if (!map.containsKey(key)) {
-        throw new RuntimeException(String.format("Key '%s' not found in map.", key));
-      }
-      final Object object = map.get(key);
-      if (object != null) {
-        getObjectMapper().readValue(input.getValue(), object.getClass());
-      }
-    }
+    // Otherwise just use the input list (even if it's empty).
+    return new FlightMap(inputList);
   }
 
   /** Convert the map to an unmodifiable form. */
   public void makeImmutable() {
     map = Collections.unmodifiableMap(map);
   }
 
+  /** Check map for emptiness */
+  boolean isEmpty() {
+    return map.isEmpty();
+  }
+
   /**
-   * Return the object from the hash map cast to the right type. Throw an exception if the Object
-   * cannot be cast to that type.
+   * Return the object from the hash map deserialized to the right type. Throw an exception if the
+   * Object cannot be deserialized to that type.
    *
    * @param <T> - type of class to expect in the hash map
    * @param key - key to lookup in the hash map
    * @param type - class requested
    * @return null if not found
-   * @throws ClassCastException if found, not castable to the requested type
+   * @throws ClassCastException if found, not deserializable to the requested type
    */
+  @Nullable
   public <T> T get(String key, Class<T> type) {
-    Object value = map.get(key);
+    String value = map.get(key);
+
     if (value == null) {
       return null;
     }
 
-    if (type.isInstance(value)) {
-      return type.cast(value);
+    try {
+      return getObjectMapper().readValue(value, type);
+    } catch (JsonProcessingException ex) {
+      throw new ClassCastException(
```

Comment:
this doesn't seem like the right exception anymore. Is there anything that depends on this exception being thrown? (if so, that suggests to me it should be a checked exception). Should we at least propagate the underlying JsonProcessIngException?

Reply:
I searched and don't see anyone catching `ClassCastException` in this context.  So perhaps throw (unchecked) `JsonConversionException` makes more sense here? 

@ddietterich: interested in your input given your plans to rationalize exceptions in your upcoming reorg work.

Reply:
I searched and don't see anyone catching ClassCastException in this context. So perhaps throw (unchecked) JsonConversionException makes more sense here?

@ddietterich: interested in your input given your plans to rationalize exceptions in your upcoming reorg work.

Reply:
I agree that JsonConversionException makes more sense.